src/main/java/com/gtnewhorizons/angelica/glsm/debug/OpenGLDebugging.java

```diff
@@ -0,0 +1,441 @@
+        new GLproperty(GL11.GL_VERTEX_ARRAY, "GL_VERTEX_ARRAY", "Vertex array enable", "vertex-array", GL_IS_ENABLED),
+        new GLproperty(GL11.GL_VERTEX_ARRAY_POINTER, "GL_VERTEX_ARRAY_POINTER", "Pointer to the vertex array", "vertex-array", "glGetPointerv()"),
+        new GLproperty(GL11.GL_VERTEX_ARRAY_SIZE, "GL_VERTEX_ARRAY_SIZE", "Coordinates per vertex", "vertex-array", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_VERTEX_ARRAY_STRIDE, "GL_VERTEX_ARRAY_STRIDE", "Stride between vertices", "vertex-array", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_VERTEX_ARRAY_TYPE, "GL_VERTEX_ARRAY_TYPE", "Type of vertex coordinates", "vertex-array", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_VIEWPORT, "GL_VIEWPORT", "Viewport origin and extent", "viewport", GL_GET_INTEGERV),
+        new GLproperty(GL11.GL_ZOOM_X, "GL_ZOOM_X", "x zoom factor", "pixel", GL_GET_FLOATV),
+        new GLproperty(GL11.GL_ZOOM_Y, "GL_ZOOM_Y", "y zoom factor", "pixel", GL_GET_FLOATV),
+    };
+
+    public static void dumpToFile(Consumer<Consumer<String>> dumper, String fileName) {
+        final List<String> lines = new ArrayList<>();
+        dumper.accept(lines::add);
+        try {
+            FileUtils.writeLines(new File(fileName), lines);
+        } catch (IOException e) {
+            LOGGER.error("Failed to write file");
+            e.printStackTrace();
+        }
+    }
+
+    public static void dumpStateToFile() {
+        dumpToFile((p) -> dumpState(p, true), "gl-dump-state.txt");
+    }
+    public static void dumpGLSMStateToFile() {
+        dumpToFile((p) -> dumpState(p, true), "glsm-dump-state.txt");
+    }
+
+    public static void dumpState(boolean glsm) {
+        dumpState(LINE_LOGGER, glsm);
+    }
+
+    public static void dumpState(Consumer<String> printer, boolean glsm) {
+        for (int i = 0; i < instance.propertyList.length; ++i) {
+            final GLproperty gLproperty = instance.propertyList[i];
+            printer.accept(gLproperty.name + ":" + gLproperty.getAsString(glsm) + " (" + gLproperty.description + ")");
+        }
+    }
+
+    public static void dumpIsEnabledToFile() {
+        dumpToFile(OpenGLDebugging::dumpIsEnabled, "gl-dump-enabled.txt");
+    }
+
+    public static void dumpIsEnabled() {
+        dumpIsEnabled(LINE_LOGGER);
+    }
+
+    public static void dumpIsEnabled(Consumer<String> printer) {
+        for (int i = 0; i < instance.propertyList.length; ++i) {
+            if (instance.propertyList[i].fetchCommand.equals(GL_IS_ENABLED)) {
+                printer.accept(instance.propertyList[i].name + ":");
+                printer.accept("" + GL11.glIsEnabled(instance.propertyList[i].gLconstant));
+                printer.accept(" (" + instance.propertyList[i].description + ")");
+            }
+        }
+    }
+
+    public static void dumpTypeToFile(String type, boolean glsm) {
+        dumpToFile((p) -> dumpType(type, p, glsm), "gl-dump-type-" + type + ".txt");
```

Comment:
The file name should reflect the glsm-ness like in `dump(GLSM)StateToFile`.
```suggestion
        dumpToFile((p) -> dumpType(type, p, glsm), (glsm ? "glsm" : "gl") + "-dump-type-" + type + ".txt");
```